runtime/src/accounts_background_service.rs

```diff
@@ -0,0 +1,160 @@
+// Service to clean up dead slots in accounts_db
+//
+// This can be expensive since we have to walk the append vecs being cleaned up.
+
+use crate::{
+    bank::{Bank, BankSlotDelta},
+    bank_forks::{BankForks, SnapshotConfig},
+    snapshot_package::AccountsPackageSender,
+    snapshot_utils,
+};
+use crossbeam_channel::{Receiver, Sender};
+use log::*;
+use solana_measure::measure::Measure;
+use solana_sdk::clock::Slot;
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, RwLock,
+};
+use std::thread::{self, sleep, Builder, JoinHandle};
+use std::time::Duration;
+
+const INTERVAL_MS: u64 = 100;
+const SHRUNKEN_ACCOUNT_PER_SEC: usize = 250;
+const SHRUNKEN_ACCOUNT_PER_INTERVAL: usize =
+    SHRUNKEN_ACCOUNT_PER_SEC / (1000 / INTERVAL_MS as usize);
+
+pub type SnapshotRequestSender = Sender<SnapshotRequest>;
+pub type SnapshotRequestReceiver = Receiver<SnapshotRequest>;
+
+pub struct SnapshotRequest {
+    pub snapshot_root_bank: Arc<Bank>,
+    pub status_cache_slot_deltas: Vec<BankSlotDelta>,
+}
+
+pub struct SnapshotItems {
+    pub snapshot_config: SnapshotConfig,
+    pub snapshot_request_receiver: SnapshotRequestReceiver,
+    pub accounts_package_sender: AccountsPackageSender,
+}
+
+pub struct AccountsBackgroundService {
+    t_background: JoinHandle<()>,
+}
+
+impl AccountsBackgroundService {
+    pub fn new(
+        bank_forks: Arc<RwLock<BankForks>>,
+        exit: &Arc<AtomicBool>,
+        snapshot_items: Option<SnapshotItems>,
+    ) -> Self {
+        info!("AccountsBackgroundService active");
+        let exit = exit.clone();
+        let mut consumed_budget = 0;
+        let mut last_cleaned_slot = 0;
```

Comment:
nits: `last_cleaned_block_height` (well, the original code was already wrong, but why not correct, if touching the code?)